File: push/src/main/java/com/xiaomi/mipush/sdk/stat/client/EventDataItem.java
package com.xiaomi.mipush.sdk.stat.client;

import android.text.TextUtils;
import com.xiaomi.push.service.PushServiceConstants;
import java.util.HashMap;
import java.util.Map;
import java.util.Set;
import org.json.JSONException;
import org.json.JSONObject;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/mipush/sdk/stat/client/EventDataItem.class */
public class EventDataItem {
    private static final String COUNT = "count";
    private static final String EVENT = "event";
    private static final String NUMERIC = "numeric";
    private static final String PROPERTY = "property";
    private String mCategory;
    private String mKey;
    private Map<String, String> mParams = new HashMap();
    private long mTimeStamp;
    private String mType;
    private String mValue;

    public EventDataItem() {
    }

    public EventDataItem(String str, String str2, String str3, String str4) {
        this.mCategory = str;
        this.mKey = str2;
        this.mValue = str3;
        this.mType = str4;
        setTimeStamp(System.currentTimeMillis());
    }

    public EventDataItem(String str, String str2, String str3, String str4, Map<String, String> map) {
        setCategory(str);
        setKey(str2);
        setValue(str3);
        setType(str4);
        setParams(map);
        setTimeStamp(System.currentTimeMillis());
    }

    public static EventDataItem getCalculateEvent(String str, String str2, long j) {
        EventDataItem eventDataItem = new EventDataItem();
        eventDataItem.setCategory(str);
        eventDataItem.setKey(str2);
        eventDataItem.setValue(String.valueOf(j));
        eventDataItem.setType(COUNT);
        eventDataItem.setTimeStamp(System.currentTimeMillis());
        return eventDataItem;
    }

    public static EventDataItem getCalculateEvent(String str, String str2, long j, Map<String, String> map) {
        EventDataItem eventDataItem = new EventDataItem();
        eventDataItem.setCategory(str);
        eventDataItem.setKey(str2);
        eventDataItem.setValue(String.valueOf(j));
        eventDataItem.setType(COUNT);
        eventDataItem.setParams(map);
        eventDataItem.setTimeStamp(System.currentTimeMillis());
        return eventDataItem;
    }

    public static EventDataItem getCountEvent(String str, String str2) {
        EventDataItem eventDataItem = new EventDataItem();
        eventDataItem.setCategory(str);
        eventDataItem.setKey(str2);
        eventDataItem.setValue(String.valueOf(1));
        eventDataItem.setType("event");
        eventDataItem.setTimeStamp(System.currentTimeMillis());
        return eventDataItem;
    }

    public static EventDataItem getCountEvent(String str, String str2, Map<String, String> map) {
        EventDataItem eventDataItem = new EventDataItem();
        eventDataItem.setCategory(str);
        eventDataItem.setKey(str2);
        eventDataItem.setValue(String.valueOf(1));
        eventDataItem.setType("event");
        eventDataItem.setParams(map);
        eventDataItem.setTimeStamp(System.currentTimeMillis());
        return eventDataItem;
    }

    public static EventDataItem getNumericPropertyEvent(String str, String str2, long j) {
        EventDataItem eventDataItem = new EventDataItem();
        eventDataItem.setCategory(str);
        eventDataItem.setKey(str2);
        eventDataItem.setValue(String.valueOf(j));
        eventDataItem.setType(NUMERIC);
        eventDataItem.setTimeStamp(System.currentTimeMillis());
        return eventDataItem;
    }

    public static EventDataItem getPropertyEvent(String str, String str2, String str3) {
        EventDataItem eventDataItem = new EventDataItem();
        eventDataItem.setCategory(str);
        eventDataItem.setKey(str2);
        eventDataItem.setValue(str3);
        eventDataItem.setType(PROPERTY);
        eventDataItem.setTimeStamp(System.currentTimeMillis());
        return eventDataItem;
    }

    private static String mapToJson(Map<String, String> map) {
        JSONObject jSONObject = new JSONObject();
        if (map != null && map.size() > 0) {
            try {
                Set<String> setKeySet = map.keySet();
                if (setKeySet != null) {
                    for (String str : setKeySet) {
                        if (!TextUtils.isEmpty(str)) {
                            jSONObject.put(str, map.get(str) + "");
                        }
                    }
                }
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
        return jSONObject.toString();
    }

    public void setCategory(String str) {
        this.mCategory = str;
    }

    public void setKey(String str) {
        this.mKey = str;
    }

    public void setParams(Map<String, String> map) {
        if (map == null || map.size() <= 0) {
            return;
        }
        this.mParams.putAll(map);
    }

    public void setTimeStamp(long j) {
        this.mTimeStamp = j;
    }

    public void setType(String str) {
        this.mType = str;
    }

    public void setValue(String str) {
        this.mValue = str;
    }

    public JSONObject toJson() {
        JSONObject jSONObject = new JSONObject();
        try {
            if (!TextUtils.isEmpty(this.mCategory)) {
                jSONObject.put("category", this.mCategory);
            }
            if (!TextUtils.isEmpty(this.mKey)) {
                jSONObject.put("key", this.mKey);
            }
            if (!TextUtils.isEmpty(this.mValue)) {
                jSONObject.put(PushServiceConstants.EXTENSION_ATTRIBUTE_APP_PARAM_VALUE, this.mValue);
            }
            if (!TextUtils.isEmpty(this.mType)) {
                jSONObject.put("type", this.mType);
            }
            Map<String, String> map = this.mParams;
            if (map != null && map.size() > 0) {
                jSONObject.put("params", mapToJson(this.mParams));
            }
            jSONObject.put("timeStamp", this.mTimeStamp);
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return jSONObject;
    }
}

File: legacy/src/main/java/com/xiaomi/mipush/sdk/stat/client/EventDataItem.kt
package com.xiaomi.mipush.sdk.stat.client
import io.github.magisk317.mipush.protocol.model.*

import android.text.TextUtils
import org.json.JSONException
import org.json.JSONObject

class EventDataItem {
    private var mCategory: String? = null
    private var mKey: String? = null
    private val mParams: MutableMap<String, String> = HashMap()
    private var mTimeStamp: Long = 0
    private var mType: String? = null
    private var mValue: String? = null

    constructor()

    constructor(
        str: String,
        str2: String,
        str3: String,
        str4: String,
    ) {
        mCategory = str
        mKey = str2
        mValue = str3
        mType = str4
        timeStamp = System.currentTimeMillis()
    }

    constructor(
        str: String,
        str2: String,
        str3: String,
        str4: String,
        map: Map<String, String>,
    ) {
        category = str
        key = str2
        value = str3
        type = str4
        setParams(map)
        timeStamp = System.currentTimeMillis()
    }

    companion object {
        private const val COUNT = "count"
        private const val EVENT = "event"
        private const val NUMERIC = "numeric"
        private const val PROPERTY = "property"

        fun getCalculateEvent(str: String, str2: String, j: Long): EventDataItem {
            return EventDataItem().apply {
                category = str
                key = str2
                value = j.toString()
                type = COUNT
                timeStamp = System.currentTimeMillis()
            }
        }

        fun getCalculateEvent(
            str: String,
            str2: String,
            j: Long,
            map: Map<String, String>,
        ): EventDataItem {
            return EventDataItem().apply {
                category = str
                key = str2
                value = j.toString()
                type = COUNT
                setParams(map)
                timeStamp = System.currentTimeMillis()
            }
        }

        fun getCountEvent(str: String, str2: String): EventDataItem {
            return EventDataItem().apply {
                category = str
                key = str2
                value = "1"
                type = EVENT
                timeStamp = System.currentTimeMillis()
            }
        }

        fun getCountEvent(
            str: String,
            str2: String,
            map: Map<String, String>,
        ): EventDataItem {
            return EventDataItem().apply {
                category = str
                key = str2
                value = "1"
                type = EVENT
                setParams(map)
                timeStamp = System.currentTimeMillis()
            }
        }

        fun getNumericPropertyEvent(str: String, str2: String, j: Long): EventDataItem {
            return EventDataItem().apply {
                category = str
                key = str2
                value = j.toString()
                type = NUMERIC
                timeStamp = System.currentTimeMillis()
            }
        }

        fun getPropertyEvent(str: String, str2: String, str3: String): EventDataItem {
            return EventDataItem().apply {
                category = str
                key = str2
                value = str3
                type = PROPERTY
                timeStamp = System.currentTimeMillis()
            }
        }

        private fun mapToJson(map: Map<String, String>): String {
            val jSONObject = JSONObject()
            if (map.isNotEmpty()) {
                try {
                    for (str in map.keys) {
                        if (!TextUtils.isEmpty(str)) {
                            jSONObject.put(str, "${map[str]}")
                        }
                    }
                } catch (e: Exception) {
                    e.printStackTrace()
                }
            }
            return jSONObject.toString()
        }
    }

    var category: String?
        get() = mCategory
        set(value) {
            mCategory = value
        }

    var key: String?
        get() = mKey
        set(value) {
            mKey = value
        }

    fun setParams(map: Map<String, String>) {
        if (map.isEmpty()) return
        mParams.putAll(map)
    }

    var timeStamp: Long
        get() = mTimeStamp
        set(value) {
            mTimeStamp = value
        }

    var type: String?
        get() = mType
        set(value) {
            mType = value
        }

    var value: String?
        get() = mValue
        set(value) {
            mValue = value
        }

    fun toJson(): JSONObject {
        val jSONObject = JSONObject()
        try {
            if (!TextUtils.isEmpty(mCategory)) {
                jSONObject.put("category", mCategory)
            }
            if (!TextUtils.isEmpty(mKey)) {
                jSONObject.put("key", mKey)
            }
            if (!TextUtils.isEmpty(mValue)) {
                jSONObject.put("value", mValue)
            }
            if (!TextUtils.isEmpty(mType)) {
                jSONObject.put("type", mType)
            }
            if (mParams.isNotEmpty()) {
                jSONObject.put("params", mapToJson(mParams))
            }
            jSONObject.put("timeStamp", mTimeStamp)
        } catch (e: JSONException) {
            e.printStackTrace()
        }
        return jSONObject
    }
}
